packages/common/src/authentication/AuthContext.tsx

```diff
@@ -99,8 +106,12 @@ export const AuthProvider: FC = ({ children }) => {
   const [localStore, setLocalStore] = useState<Store | undefined>(cookieStore);
   const [localToken, setLocalToken] = useState<string | undefined>(cookieToken);
   const storeId = localStore?.id ?? '';
-
-  useRefreshingAuth(localToken);
+  const saveToken = (token?: string) => {
+    setLocalToken(token);
+    const authCookie = getAuthCookie();
+    setAuthCookie({ ...authCookie, token: token ?? '' });
```

Comment:
Working nicely!

-----
Stop reading
-----
Random thought dump

 I had thoughts around abstracting some of this away into a `useAuthCookie` hook.. maybe using zustand so it could be used in multiple places (useAuthToken?) but I am not sure it's worth it. 

For example

```
const useAuthCookie = zustand({ ... })
const { store, token, setStore, setToken, setCookie } = useAuthCookie()

useAuthCookie.subscribe(() => {
    // listen for changes and update the actual cookie in the browser
})
```

Though, since there's also the `AuthContext`.. maybe there's a nice way where instead of the `useAuthCookie`, just adding a subscriber to that state to update the cookie... maybe even adding https://www.npmjs.com/package/zustand-persist (Oh, you're still reading? I did stay stop!)

I'm not sure, but I think the store is in a `useState`, in the Context object, in the MRU creds, in a cookie and in the bearer token set in the auth header and there has to be a way to consolidate some of it

Reply:
oh, that's good. a fine idea - actually a couple of good ideas in there. will pop this pr in and make an issue